src/code/z_DLF.c

```diff
@@ -2,6 +2,78 @@
 #include "system_malloc.h"
 #include "z64load.h"
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_DLF/Overlay_LoadGameState.s")
+void Overlay_LoadGameState(GameStateOverlay* gameState) {
+    void* temp;
+    void* vrStart;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_DLF/Overlay_FreeGameState.s")
+    if (gameState->loadedRamAddr != NULL) {
+        return;
+    }
+    vrStart = gameState->vramStart;
+    if (vrStart == NULL) {
+        gameState->unk_28 = 0;
+    } else {
+        gameState->loadedRamAddr = Load2_AllocateAndLoad(gameState->vromStart, gameState->vromEnd, (uintptr_t)vrStart,
+                                                         (uintptr_t)gameState->vramEnd);
+        if (gameState->loadedRamAddr == NULL) {
+            return;
+        }
+
+        gameState->unk_14 = (uintptr_t)((gameState->unk_14 != NULL) ? (void*)(-OVERLAY_RELOCATION_OFFSET(gameState) + (uintptr_t)gameState->unk_14) : NULL);
+        gameState->init = (uintptr_t)((gameState->init != NULL) ? (void*)(-OVERLAY_RELOCATION_OFFSET(gameState) + (uintptr_t)gameState->init) : NULL);
+        gameState->destroy = (uintptr_t)((gameState->destroy != NULL) ? (void*)(-OVERLAY_RELOCATION_OFFSET(gameState) + (uintptr_t)gameState->destroy) : NULL);
+        gameState->unk_20 = (uintptr_t)((gameState->unk_20 != NULL) ? (void*)(-OVERLAY_RELOCATION_OFFSET(gameState) + (uintptr_t)gameState->unk_20) : NULL);
+        gameState->unk_24 = (uintptr_t)((gameState->unk_24 != NULL) ? (void*)(-OVERLAY_RELOCATION_OFFSET(gameState) + (uintptr_t)gameState->unk_24) : NULL);
+
+        gameState->unk_28 = 0;
+    }
+}
+
+// This is probably a fake match
+// https://decomp.me/scratch/qZKzC
+
+void Overlay_FreeGameState(GameStateOverlay* gameState) {
+    s32 var_v0;
+    void* temp;
```

Comment:
this temp and comment can go once the solution is updated:
https://decomp.me/scratch/Ok1Bj

Reply:
To add onto this, I think it would be best if `Overlay_LoadGameState`, as well as the instances in `Actor_LoadOverlay` and `EffectSs_Spawn` be changed to match this style, i.e. move the `OVERLAY_RELOCATION_OFFSET` macro to the right side with an `intptr_t` cast.